### ISSUE TITLE
bug: Refocusing on VS Code doesn't update Cody to the correct Free|Pro state

### ISSUE THREAD

**Issue by chenkc805** [sourcegraph/cody#2220]
### Version

v0.19.1702047982

### Describe the bug

Expected behavior:
- When refocusing on the VS Code window (through any method: alt+tab, clicking over, mission control+clicking over), Cody should update to the correct state the user is in, either the Free tier or Pro tier.

I tried 4 different scenarios, and sometimes they work and sometimes they don't:
- Start on Free --> Hit upgrade button in VSCode --> Upgrade to Pro in web --> click back over to VS Code --> ✅ Cody successfully refreshes to Pro
- Start on Pro --> alt+tab over to the Web --> Downgrade to Free in web --> click back over to VS Code --> ❌  Cody doesn't  refreshes to Free
- Start on Free --> Hit upgrade button in VSCode --> Upgrade to Pro in web --> alt+tab over to VS Code --> ✅ Cody successfully refreshes to Pro
- Start on Pro --> alt+tab over to the Web --> Downgrade to Free in web --> alt+tab over to VS Code --> ✅ Cody doesn't  refreshes to Free

What invariably works is restarting VS Code. CC: @toolmantim @danielmarquespt let's think of some light copy that we can use to tell users to restart VS Code if they're not seeing their upgrade/downgrade happen successfully 

Loom here: https://sourcegraph.slack.com/archives/C066F85TE80/p1702054424002389

### Expected behavior

Expected behavior:
- When refocusing on the VS Code window (through any method: alt+tab, clicking over, mission control+clicking over), Cody should update to the correct state the user is in, either the Free tier or Pro tier.

**Comment by toolmantim** [comment]
If we already support this:

> Start on Free --> Hit upgrade button in VSCode --> Upgrade to Pro in web --> click back over to VS Code --> ✅ Cody successfully refreshes to Pro

then I think the following might be a bug we could just address in VS Code:

> Start on Pro --> alt+tab over to the Web --> Downgrade to Free in web --> click back over to VS Code --> ❌ Cody doesn't refreshes to Free

but if it doesn’t get fixed for GA, some copy would help.

**Comment by chenkc805** [comment]
@toolmantim can I get your help with the copy fix here: https://github.com/sourcegraph/cody/issues/2277 assigned to you

**Comment by toolmantim** [comment]
I can, but can you still repro this on the latest version? I tested these and they all seemed to work.

**Comment by toolmantim** [comment]
I'm marking this as P2 for now until we can consistently repro on latest

**Comment by toolmantim** [comment]
If we can repro it, there's a few options for solving this, including added a UI button to "Refresh Account Status" in VS Code.

The new "Account" sidebar nav item that shows the current plan at least makes this much more obvious, leading users to know "Hmm that's not right. Let me restart VS Code" (which is something that users often do in user testing)

**Comment by chenkc805** [comment]
Can't repro this at the moment. It's been pretty spotty and not sure why it sometimes works and sometimes doesn't.

**Comment by DanTup** [comment]
I tried (with various combinations of Alt+Tab, clicking the window, closing Chrome so it automatically become active) but everything seemed to work correctly.

I did find [this interesting issue](https://github.com/microsoft/vscode/issues/113539) about it not working if you clicked into a webview editor, however it's been fixed and that didn't trigger it for me either. There's also [this one](https://github.com/microsoft/vscode/issues/149487) but it may be limited to some specific Linux window managers (@chenkc805 out of interest, what platform were you using?).

Could it be related to having multiple accounts? I notice when I click "Account" in the sidebar, there's a "Switch Account" option, but since I only have one it's not clear to me exactly how this works. It's also perhaps not guaranteed that the "current" account in VS Code is the same one that is logged in in the browser window that is spawned? (although, none of these explain why it would appear to work some times if you were testing them all within the same session).

I also wonder whether any error is logged here when it occurs (could the graphql request be failing?):

https://github.com/sourcegraph/cody/blob/2a00da01ae8b83ecd5450af7be2682c32edf3080/vscode/src/main.ts#L510-L514
